litestar/dto/_codegen_backend.py

```diff
@@ -0,0 +1,576 @@
+            field_definition: Parsed type.
+            handler_id: The name of the handler that this backend is for.
+            is_data_field: Whether the field is a subclass of DTOData.
+            model_type: Model type.
+            wrapper_attribute_name: If the data that DTO should operate upon is wrapped in a generic datastructure,
+              this is the name of the attribute that the data is stored in.
+        """
+        super().__init__(
+            dto_factory=dto_factory,
+            field_definition=field_definition,
+            handler_id=handler_id,
+            is_data_field=is_data_field,
+            model_type=model_type,
+            wrapper_attribute_name=wrapper_attribute_name,
+        )
+        self._transfer_to_dict = self._create_transfer_data_fn(
+            destination_type=dict,
+            field_definition=self.field_definition,
+        )
+        self._transfer_to_model_type = self._create_transfer_data_fn(
+            destination_type=self.model_type,
+            field_definition=self.field_definition,
+        )
+        self._transfer_data_from_builtins = self._create_transfer_data_fn(
+            destination_type=self.model_type,
+            field_definition=self.field_definition,
+            override_serialization_name=False,
+        )
+        self._transfer_data_from_builtins_with_overrides = self._create_transfer_data_fn(
+            destination_type=self.model_type,
+            field_definition=self.field_definition,
+            override_serialization_name=True,
+        )
+        self._encode_data = self._create_transfer_data_fn(
+            destination_type=self.transfer_model_type,
+            field_definition=self.field_definition,
+        )
+
+    def populate_data_from_builtins(self, builtins: Any, asgi_connection: ASGIConnection) -> Any:
+        """Populate model instance from builtin types.
+
+        Args:
+            builtins: Builtin type.
+            asgi_connection: The current ASGI Connection
+
+        Returns:
+            Instance or collection of ``model_type`` instances.
+        """
+        if self.dto_data_type:
+            return self.dto_data_type(
+                backend=self,
+                data_as_builtins=self._transfer_to_dict(self.parse_builtins(builtins, asgi_connection)),
+            )
+        return self.transfer_data_from_builtins(self.parse_builtins(builtins, asgi_connection))
+
+    def transfer_data_from_builtins(self, builtins: Any, override_serialization_name: bool = False) -> Any:
+        """Populate model instance from builtin types.
+
+        Args:
+            builtins: Builtin type.
+            override_serialization_name: Use the original field names, used when creating
+                                         an instance using `DTOData.create_instance`
+
+        Returns:
+            Instance or collection of ``model_type`` instances.
+        """
+        if override_serialization_name:
```

Comment:
Need to function here, since in the original we do a not so nice workaround where we temporarily alter an instance attribute.